contracts/BlendSwap.sol

```diff
@@ -49,6 +49,7 @@ contract BlendSwap {
     }
 
     function lock(
+        address from,
```

Comment:
Ok, imagine I decide to go with separate approve and lock steps (your changes allow this since `lock` is not guarded with `onlyBlend`). I first go and approve the bridge to lock some amount of tokens on my behalf. Then I emit a transaction to actually `lock` my funds...

... and here my heart breaks because my funds are stolen:
1) someone frontruns me and provides my `from` address but different lock params: `confirmed: true, secretHash: sha256("pwned")`
2) he then calls `redeem("pwned")`

boom

Reply:
I agree that we should at least make a separate `lock` endpoint that will only allow calls from Blend contract with `from` specified.

Reply:
I think it will be a legit solution. There is only one concern which is that Blend contract will try to use `lock` without intent from the user. This becomes a realistic attack if either:

1. Blend token will be updated with malicious code
2. Blend token would allow custom callbacks

In the former case, attacker can simply make a Blend transfer, w/o swap through Bridge.
Latter case is mitigated by calling callbacks from a separate subcontract (which will have a different address and thus doesn't allow impersonation).

So yep, let's:

1. Make a `lockFrom` function to be called either by the Blend token or from within Bridge contract.
2. Make a `lock` function to be called by anyone which subsequently calls `lockFrom` with `from := msg.sender`.